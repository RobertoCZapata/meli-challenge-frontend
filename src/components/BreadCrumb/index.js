import React from "react";
import { useContext } from "react";
import AppContext from "../../context/AppContext";

const Breadcrumb = (props) => {
  const { filters } = useContext(AppContext);

  const paths = filters[0]?.values[0]?.path_from_root;

  return (
    <div>
      <nav className="breadcrumb__container">
        <ul className="breadcrumbs">
          {paths?.map((el, index) => {
            return (
              <li key={index} className="breadcrumbs__item">
                <a href="" className="breadcrumbs__link">
                  {el.name}
                </a>
              </li>
            );
          })}
        </ul>
      </nav>
    </div>
  );
};

export default Breadcrumb;