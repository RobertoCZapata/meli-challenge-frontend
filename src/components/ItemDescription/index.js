import React from "react";
import Image from "next/image";
import { formatAsCurrency } from "../../utils/helpers";

export const ItemDescription = ({
  sold_quantity,
  title,
  image,
  description,
  price,
}) => {
  return (
    <div className="detail">
      <div className="detail__container">
        <div className="detail__image">
          {image && (
            <Image src={image} width={200} height={200} alt="Image Detail" />
          )}
        </div>
        <div className="detail__header">
          <div className="detail__subtitle">
            <span>Nuevo | {sold_quantity} vendidos</span>
          </div>
          <div className="detail__titleContainer">
            <p>{title}</p>
          </div>
          <div className="detail__price">
            <span> $ {formatAsCurrency(price)}</span>
          </div>
          <div className="detail__actions">
            <button className="detail__buttons">Comprar</button>
          </div>
        </div>
      </div>
      <div className="detail__description">
        <h2 className="detail__descriptionTitle">Descripcion del producto</h2>
        <p className="detail__descriptionText">{description}</p>
      </div>
    </div>
  );
};
