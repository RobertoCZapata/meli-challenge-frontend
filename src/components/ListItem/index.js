import React from 'react';
import Link from 'next/link';
import { formatAsCurrency } from '../../utils/helpers';
import Image from 'next/image';
import Shipping from '../../../assets/img/ic_shipping.png';
import {
  Card,
  ImageWrapper,
  PriceWrapper,
  CardContent,
  CartTitle,
  Badge,
} from './styled';

export const ListItem = ({
  id,
  title,
  price,
  image,
  freeShipping,
  condition,
}) => {
  return (
    <Card>
      <ImageWrapper>
        <Link
          href={{
            pathname: '/items/[id]',
            query: { id },
          }}
        >
          <a>
            <Image
              src={image}
              width={160}
              height={160}
              alt='Image Product'
              style={{ objectFit: 'contain', width: '100%' }}
            />
          </a>
        </Link>
      </ImageWrapper>
      <CardContent>
        <CartTitle>
          <Link
            href={{
              pathname: '/items/[id]',
              query: { id },
            }}
          >
            {title}
          </Link>
        </CartTitle>
        <PriceWrapper>
          <p>$ {formatAsCurrency(price)}</p>
          <span>
            {freeShipping && <Image src={Shipping} alt='Free Shipping' />}
          </span>
        </PriceWrapper>
        {condition === 'new' && <Badge>{condition}</Badge>}
      </CardContent>
    </Card>
  );
};
