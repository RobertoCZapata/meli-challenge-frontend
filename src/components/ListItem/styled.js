import styled from 'styled-components';

export const Card = styled.div`
  background-color: #fff;
  width: 100%;
  padding: 20px;
  display: flex;
  border-bottom: thin solid #eee;
`;

export const ImageWrapper = styled.div`
  width: 160px;
  height: 160px;
`;

export const PriceWrapper = styled.div`
  display: flex;
  align-items: center;

  p {
    font-size: 24px;
    color: #333;
    font-weight: 400;
    line-height: 1;
  }

  span {
    margin-left: 10px;
  }
`;

export const CartTitle = styled.h2`
  color: #333;
  font-size: 20px;
  font-weight: 300;
  line-height: 1.3;
  margin: 0;
`;

export const CardContent = styled.div`
  padding-top: 20px;
  padding-left: 10px;
  flex: 1;
`;

export const Badge = styled.span`
  background-color: #c6f6d5;
  text-transform: uppercase;
  padding: 5px;
  font-size: 12px;
  border-radius: 5px;
`;
