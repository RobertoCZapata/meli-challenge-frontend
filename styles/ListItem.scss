@import "variables";

.search {
  width: 100%;
  display: flex;
  flex-direction: column;
  margin: auto;

  &__wrapper {
    display: flex;
    flex-direction: row;
    padding: 20px 50px 20px 0;
    width: 100%;
    border-bottom: thin solid $color-borderbtn;
    background-color: $color-background;
    flex-grow: 1;
  }

  &__image {
    box-sizing: content-box;
    flex-shrink: 0;
    overflow: hidden;
    padding: 0 24px;
    width: 160px;
    margin: 0;
  }

  &__shipping {
    margin-left: 10px;
  }

  &__locationWrapper {
    width: 100%;
  }

  &__contentWrapper {
    display: flex;
    flex-direction: column;
  }

  &__itemPrice {
    color: $color-itemColor;
    font-size: $font-lg;
    font-weight: 400;
    line-height: 1.25;
    margin-top: 5px;
    margin-bottom: 10px;
  }

  &__itemTitle {
    color: $color-itemColor;
    font-size: $font-lg;
    font-weight: 300;
    line-height: 1.3;
    margin: 0;
  }

  &__locationTitle {
    color: $color-itemColor;
    font-size: $font-md;
    font-weight: 300;
    line-height: 1;
    text-align: end;
    padding-top: 15px;
  }

  &__desktopLocation {
    color: $color-itemColor;
    font-size: $font-md;
    font-weight: 300;
    line-height: 1;
    display: none;
  }
}

@media (max-width: 600px) {
  .search__locationWrapper {
    display: none;
  }
  .search__desktopLocation {
    display: block;
  }
}
