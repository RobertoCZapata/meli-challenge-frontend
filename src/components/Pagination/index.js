import React, { useState } from 'react';
import { StyledPagination, PaginationWrapper } from './styled';
import { useAppContext } from '../../context/AppContext';

export const Pagination = () => {
  const { getItems, page, changePage } = useAppContext();

  const handleClick = (action) => {
    const query = localStorage.getItem('query');
    let newPage = 0;

    if (action === 'next') {
      newPage = page + 1;
    } else if ('back') {
      newPage = page - 1;
    }

    changePage(newPage);
    getItems(query, newPage);
  };

  const showBackButton = page > 0;

  return (
    <PaginationWrapper>
      {showBackButton && (
        <StyledPagination onClick={() => handleClick('back')}>
          Atras
        </StyledPagination>
      )}

      <StyledPagination onClick={() => handleClick('next')}>
        Siguiente
      </StyledPagination>
    </PaginationWrapper>
  );
};