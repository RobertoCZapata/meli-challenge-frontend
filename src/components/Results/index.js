import React from 'react';
import { ResultsWrapper, SpinnerWrapper } from './styled';
import { useAppContext } from '../../context/AppContext';
import { Spinner } from '../Spinner';
import PropTypes from 'prop-types';

export const Results = ({ children }) => {
  const { isLoading } = useAppContext();

  return (
    <ResultsWrapper>
      {isLoading ? (
        <SpinnerWrapper>
          <Spinner />
        </SpinnerWrapper>
      ) : (
        children
      )}
    </ResultsWrapper>
  );
};

Results.propTypes = {
  children: PropTypes.node,
};