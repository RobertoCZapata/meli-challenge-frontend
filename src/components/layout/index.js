import React, { useContext } from 'react';
import Head from 'next/head';

import { SearchBar } from '../SearchBar/index';
import Breadcrumb from '../BreadCrumb/index';
import { Header, Container } from './styled';

export const Layout = ({ children }) => {
  return (
    <div>
      <Head>
        <title>Meli App</title>
        <meta name='description' content='Generated by create next app' />
        <link rel='icon' href='/favicon.ico' />
      </Head>

      <Header>
        <SearchBar />
      </Header>
      <Container>
        {/* <Breadcrumb /> */}
        {children}
      </Container>
    </div>
  );
};
