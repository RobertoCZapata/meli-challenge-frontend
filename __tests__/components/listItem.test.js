import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { ListItem } from '../../src/components/ListItem';

describe('ListItem', () => {
  it('should render list item component', () => {
    render(
      <ListItem
        id='1'
        title='some title'
        price='10000'
        freeShipping={true}
        image='http://http2.mlstatic.com/D_820151-MLA50232582674_062022-I.jpg'
        condition='new'
      />
    );

    screen.getByText('new');
    screen.getByText('some title');
    screen.getByText('$ 10.000');
  });

  it('should match snapshot', () => {
    const tree = renderer
      .create(
        <ListItem
          id='1'
          title='some title'
          price='10000'
          freeShipping={true}
          image='http://http2.mlstatic.com/D_820151-MLA50232582674_062022-I.jpg'
          condition='new'
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
