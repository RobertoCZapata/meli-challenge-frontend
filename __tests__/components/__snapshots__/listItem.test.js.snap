// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should match snapshot 1`] = `
<div
  className="sc-bcXHqe iUhTtb"
>
  <div
    className="sc-gswNZR eXNMqo"
  >
    <a
      href="/items/1"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <span
        style={
          {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "inline-block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "width": "initial",
          }
        }
      >
        <span
          style={
            {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        >
          <img
            alt=""
            aria-hidden={true}
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27160%27%20height=%27160%27/%3e"
            style={
              {
                "background": "none",
                "border": 0,
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          />
        </span>
        <img
          alt="Image Product"
          data-nimg="intrinsic"
          decoding="async"
          onError={[Function]}
          onLoad={[Function]}
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
        <noscript>
          <img
            alt="Image Product"
            data-nimg="intrinsic"
            decoding="async"
            loading="lazy"
            src="/_next/image?url=http%3A%2F%2Fhttp2.mlstatic.com%2FD_820151-MLA50232582674_062022-I.jpg&w=384&q=75"
            srcSet="/_next/image?url=http%3A%2F%2Fhttp2.mlstatic.com%2FD_820151-MLA50232582674_062022-I.jpg&w=256&q=75 1x, /_next/image?url=http%3A%2F%2Fhttp2.mlstatic.com%2FD_820151-MLA50232582674_062022-I.jpg&w=384&q=75 2x"
            style={
              {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </noscript>
      </span>
    </a>
  </div>
  <div
    className="sc-eDvSVe lkSCjN"
  >
    <h2
      className="sc-hLBbgP irYjXH"
    >
      <a
        href="/items/1"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        some title
      </a>
    </h2>
    <div
      className="sc-dkrFOg clTdbK"
    >
      <p>
        $ 
        10.000
      </p>
      <span>
        <span
          style={
            {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "width": "initial",
            }
          }
        >
          <span
            style={
              {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          >
            <img
              alt=""
              aria-hidden={true}
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
              style={
                {
                  "background": "none",
                  "border": 0,
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            />
          </span>
          <img
            alt="Free Shipping"
            data-nimg="intrinsic"
            decoding="async"
            onError={[Function]}
            onLoad={[Function]}
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
          <noscript>
            <img
              alt="Free Shipping"
              data-nimg="intrinsic"
              decoding="async"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
              style={
                {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </noscript>
        </span>
      </span>
    </div>
    <span
      className="sc-jSUZER dmBCMl"
    >
      new
    </span>
  </div>
</div>
`;
